src/transformers/trainer.py

```diff
@@ -635,8 +635,7 @@ def __init__(
                         f"FP16 provided in SM_HP_MP_PARAMETERS is {smp.state.cfg.fp16}, "
                         "but SageMaker Model Parallelism < 1.10 does not support FP16 in trainer."
                     )
-
-        if (args.fp16 or args.bf16) and self.sharded_ddp is not None:
+        if args.fp16 or args.bf16:
```

Comment:
Now, `mixed precision` handling is moved to Accelerate. Only when Fairscale is being used mixed precision handling is supported in trainer as it isn't present in Accelerate.

Reply:
So, this change can't be done